### PR TITLE
fix: Filter eventTimer properties with undefined values

### PR DESCRIPTION
**Pull request by chrislomaxjones**
## What does this change?

Filters event timer properties with undefined values.

~Also introduces a `PropertyEntry` type and annotation in order to prevent a [no-unnecessary-condition
](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md) eslint errror.~

## Why?

Prevents undefined properties being included and sent with commercial metrics.

### CODE REVIEW

**Comment by mxdvl** [comment]
Hey @chrislomaxjones, I'm not sure why we need to introduce a new type. I think the properties type should take care of this.

Maybe we need to refactor a little, as `downlink` and `effectiveType` are only available on Chrome.

cc. @zekehuntergreen

**Comment by chrislomaxjones** [comment]
@mxdvl I get an eslint error if the type of the argument isn't annotated (see screenshot below). There may be a better solution to this though as the types do overlap!

Screenshot 2021-06-09 at 08 24 33

File: src/sendCommercialMetrics.spec.ts [hunk 1]
```diff
@@ -19,39 +44,108 @@ describe('sendCommercialMetrics', () => {
...
-				JSON.stringify({
-					browser_id: 'browser id',
-					page_view_id: 'page view id',
-					received_timestamp: '2021-01-01T12:00:00.000Z',
-					received_date: '2021-01-01',
-					platform: 'NEXT_GEN',
-					metrics: [],
-					properties: [{ name: 'isDev', value: 'localhost' }],
-				}),
+				'//performance-events.guardianapis.com/commercial-metrics',
```
Comment (mxdvl) [inline src/sendCommercialMetrics.spec.ts:53]: We could extract this in a constant, like `PROD_ENDPOINT`.

**Review by mxdvl — APPROVED** [review]
Increasing coverage, excellent 🎉!

File: src/sendCommercialMetrics.spec.ts (hunk 1 above)
Comment (zekehuntergreen) [reply src/sendCommercialMetrics.spec.ts]: Non-blocking: might be worth doing the same here? https://github.com/guardian/commercial-core/pull/267/files#diff-c076222fcae3be8e833dc2a29ce1870cce642fe2aab8c5504875182b46de95c6R34

**Review by zekehuntergreen — APPROVED** [review]

File: src/sendCommercialMetrics.spec.ts (hunk 1 above)
Comment (mxdvl) [reply src/sendCommercialMetrics.spec.ts]: Yes, and then exporting that `const` to reuse it in the test!

**Comment by mxdvl** [comment]
Next up: raise a PR in `frontend` and `dotcom-rendering` to update to the latest!

**Pull request merged**